### PR TITLE
NIAD-363: Deployment and configuration of fake-mesh into vp cluster

### PR DESCRIPTION
**Pull request by mattd-kainos**

### CODE REVIEW

File: terraform/aws/components/nhais/locals_env_variables.tf [hunk 1]
```diff
@@ -46,7 +46,8 @@ locals {
     },
     {
       name = "NHAIS_MESH_HOST"
-      value = var.nhais_mesh_host //https://fake-mesh:8829/messageexchange/ / https://msg.intspineservices.nhs.uk/
+      // TODO: hardcoded default for now but it should just be from the var. This will be different for every env
+      value = var.nhais_mesh_host == "" ? "https://mesh.vp.nhsredteam.internal.nhs.uk:8829/messageexchange/" : var.nhais_mesh_host
```
Comment (LukaszKwasniewski) [inline terraform/aws/components/nhais/locals_env_variables.tf]: make the url depend on environment, replace vp with var.environment

**Review by LukaszKwasniewski — DISMISSED** [review]

File: terraform/aws/components/fake_mesh/data_remote_state_account.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+data "terraform_remote_state" "account" {
+  backend = "s3"
+  
+  config = {
+    bucket = var.tf_state_bucket
+    key = "${var.project}-account-account.tfstate"
```
Comment (grzegorz-robak) [inline terraform/aws/components/fake_mesh/data_remote_state_account.tf:6]: other files use `${var.project}-${var.environment}`, is above correct?

File: terraform/aws/components/fake_mesh/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,4 @@
+//output "nhais_fake_mesh_security_group" {
+//  value = module.fake_mesh_ecs_service.service_sg_id
+//  description = "SG id of the fake-mesh container"
+//}
```
Comment (grzegorz-robak) [inline terraform/aws/components/fake_mesh/outputs.tf]: commented file?

File: terraform/aws/modules/module_ecs_service/variables.tf [hunk 4]
```diff
@@ -202,6 +202,18 @@ variable "load_balancer_type" {
...
 
+variable "use_application_lb" {
+  type = bool
+  description = "True if the LB should be application type"
+  default = true
+}
+
+variable "use_network_lb" {
+  type = bool
+  description = "True if the LB should be application type"
```
Comment (grzegorz-robak) [inline terraform/aws/modules/module_ecs_service/variables.tf]: network, not application

**Review by grzegorz-robak — CHANGES_REQUESTED** [review]

File: terraform/aws/modules/module_ecs_service/variables.tf (hunk 4 above)
Comment (mattd-kainos) [reply terraform/aws/modules/module_ecs_service/variables.tf]: changed description

File: terraform/aws/components/fake_mesh/outputs.tf (hunk 3 above)
Comment (mattd-kainos) [reply terraform/aws/components/fake_mesh/outputs.tf]: removed file

**Review by jaff47 — DISMISSED** [review]

File: terraform/aws/components/nhais/locals_env_variables.tf (hunk 1 above)
Comment (mattd-kainos) [reply terraform/aws/components/nhais/locals_env_variables.tf]: fake-mesh should only ever be deployed into the vp cluster. I suppose this default value shouldn't be here at all and it should be specified at deployment.

Comment (mattd-kainos) [reply terraform/aws/components/nhais/locals_env_variables.tf]: actually, if we want to use a default it should be in the variables.tf right?

File: terraform/aws/modules/module_ecs_service/sg_rules.tf [hunk 5]
```diff
@@ -37,3 +37,35 @@ resource "aws_security_group_rule" "healthcheck_to_service_from_load_balancer" {
   security_group_id = aws_security_group.service_sg.id
   source_security_group_id = aws_security_group.service_lb_sg.id  
 }
+
+# Healthcheck rules in case of Network LB
+
+# resource "aws_security_group_rule" "healthcheck_from_netwrok_load_balancer_to_service" {
```
Comment (mattd-kainos) [inline terraform/aws/modules/module_ecs_service/sg_rules.tf]: @LukaszKwasniewski you commented out this section. Should we leave it here?

File: terraform/aws/components/fake_mesh/data_remote_state_account.tf (hunk 2 above)
Comment (LukaszKwasniewski) [reply terraform/aws/components/fake_mesh/data_remote_state_account.tf:6]: yes, this is correct, account is specific, where both the environment and component are named 'account' as it should exists only in one copy

Comment (mattd-kainos) [reply terraform/aws/components/fake_mesh/data_remote_state_account.tf:6]: Account may be a special case. @LukaszKwasniewski ?

**Review by grzegorz-robak — DISMISSED** [review]

File: terraform/aws/components/nhais/locals_env_variables.tf (hunk 1 above)
Comment (LukaszKwasniewski) [reply terraform/aws/components/nhais/locals_env_variables.tf]: we have a default in var.nhais_mesh_host, which is empty string.
I'm aware that fake-mesh should exist only in vp, but the component code should be environment agnostic.
All environent wise settings should be done in etc/*.tfvars - deploy fake-mesh for vp, do not for others

File: terraform/aws/modules/module_ecs_service/sg_rules.tf (hunk 5 above)
Comment (LukaszKwasniewski) [reply terraform/aws/modules/module_ecs_service/sg_rules.tf]: It can be removed, meant to use it, but it turned to be not needed

**Review by jaff47 — DISMISSED** [review]

**Review by LukaszKwasniewski — APPROVED** [review]

**Pull request merged**